subsys/net/ip/Kconfig

```diff
@@ -700,6 +700,22 @@ config NET_PKT_BUF_USER_DATA_SIZE
 	help
 	  User data size used in rx and tx network buffers.
 
+config NET_PKT_ALLOW_HEADROOM_ALLOCATION
+	bool "Allow headroom allocation for L2 header"
+	depends on NET_BUF_FIXED_DATA_SIZE
+	depends on NET_L2_ETHERNET
```

Comment:
I would remove this (hard) dependency, it is currently only used for L2 ethernet, but it could be used out-of-tree.

Reply:
I only added that to emphasize that currently this is only for Ethernet but probably it is better to not hard code this. I will remove it in next version.